impeller/renderer/context.cc

```diff
@@ -16,4 +17,37 @@ bool Context::UpdateOffscreenLayerPixelFormat(PixelFormat format) {
   return false;
 }
 
+void Context::GetTextureContents(const std::shared_ptr<Texture>& texture,
+                                 const ReadbackCallback& callback) const {
+  auto cmd_buffer = CreateCommandBuffer();
+  auto blit_pass = cmd_buffer->CreateBlitPass();
+  auto buffer_size =
+      texture->GetTextureDescriptor().GetByteSizeOfBaseMipLevel();
+
+  DeviceBufferDescriptor desc;
+  desc.size = buffer_size;
+  desc.storage_mode = StorageMode::kHostVisible;
+  auto device_buffer = GetResourceAllocator()->CreateBuffer(desc);
+  if (!device_buffer) {
+    callback(nullptr);
+    return;
+  }
+
+  if (!blit_pass->AddCopy(texture, device_buffer)) {
```

Comment:
Does a texture->device buffer blit actually force a transfer from the device to the _host_? If it does, I really think that behavior should be decoupled (not now, but later). But if it doesn't force a host transfer, I don't think this patch will work?

Reply:
Also, no cap check needed here (like `SupportsBufferToTextureBlits`)?

Reply:
No, this is supported on all backends from my inspection. If it didn't work, then the test would be failing, no?

Reply:
For fixing up the blits (later), I think a good solution would be to have two different `AddCopy` overloads: One that _only_ copies to the host, and another that _only_ copies to a new device allocation. Copying to both should never be necessary.

```
  bool AddCopy(std::shared_ptr<Texture> source,
               std::vector<uint8_t> destination,
               std::optional<IRect> source_region = std::nullopt,
               IPoint destination_origin = {},
               std::string label = "");
```

Reply:
Made a follow-up issue about this here: https://github.com/flutter/flutter/issues/135783